datasync/chaindatafetcher/kafka/kafka.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2020 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package kafka
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/hashicorp/go-uuid"
+	"github.com/klaytn/klaytn/datasync/chaindatafetcher/types"
+	"github.com/klaytn/klaytn/log"
+)
+
+var (
+	logger = log.NewModuleLogger(log.ChainDataFetcher)
+	kb     *KafkaBroker
+	once   sync.Once
+)
+
+type KafkaBroker struct {
+	producer   sarama.AsyncProducer
+	admin      sarama.ClusterAdmin
```

Comment:
I have a question. Is admin a distinct server, or one of kafka brokers? Or is it like an authority anyone can aquire?
This seems weird, because it looks like anyone who can connect to kafka cluster, has admin access.

Reply:
Right. It is one of the brokers. There are settings for authorization, but let's think about this later on.